tgui/packages/tgui/interfaces/ShuttleBlueprints.tsx

```diff
@@ -0,0 +1,458 @@
+  );
+  const { act } = useBackend<ShuttleBlueprintsData>();
+  const {
+    onShuttleFrame,
+    visualizing,
+    tooManyShuttles,
+    onCustomShuttle,
+    masterExists,
+  } = props;
+  return (
+    <Stack justify="space-around">
+      <Stack.Item grow>
+        <DirectionPad
+          title="Shuttle Direction"
+          tooltip="This specifies the direction that the shuttle being built is facing."
+          enabledDirections={Direction.ALL}
+          selectedDirection={shuttleDirection}
+          onSelect={(dir) => setShuttleDirection(dir)}
+        />
+      </Stack.Item>
+      <Stack.Item>
+        <Stack fill vertical align="end" justify="space-between">
+          <Stack.Item>
+            <VisualizationToggle visualizing={visualizing} />
+          </Stack.Item>
+          <Stack.Item>
+            <Stack vertical>
+              <Stack.Item>
+                <Button.Confirm
+                  disabled={!onShuttleFrame || tooManyShuttles}
+                  tooltip={
+                    tooManyShuttles
+                      ? 'There are too many shuttles already.'
+                      : onShuttleFrame
+                        ? null
+                        : 'You must be standing on a shuttle frame to do this.'
+                  }
+                  onClick={() =>
+                    act('tryBuildShuttle', { dir: shuttleDirection })
+                  }
+                >
+                  Build New Shuttle
+                </Button.Confirm>
+              </Stack.Item>
+              <Stack.Item>
+                <Button.Confirm
+                  disabled={!onCustomShuttle || masterExists}
+                  tooltip={
+                    onCustomShuttle
+                      ? masterExists
+                        ? 'The master blueprint for this shuttle still exists. \
+                          Whoever has it can copy it to this set of blueprints.'
+                        : null
+                      : 'You must be on a custom shuttle to do this.'
+                  }
+                  onClick={() => act('tryLinkShuttle')}
+                >
+                  Connect To Existing Shuttle
+                </Button.Confirm>
+              </Stack.Item>
+            </Stack>
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const ShuttleConfiguration = (props: ShuttleConfigurationProps) => {
+  const [name, setName] = useState<string>();
+  const [mergeArea, setMergeArea] = useState<AreaData>();
+  const {
+    visualizing,
+    onShuttleFrame,
+    onShuttle,
+    inDefaultArea,
+    currentArea,
+    neighboringAreas = new Map<string, string>(),
+    apcs = new Map<string, BooleanLike>(),
+    defaultApc,
+    apcInMergeRegion,
+    idle,
+    masterExists,
+  } = props;
+  const { name: currentAreaName = '', ref: currentAreaRef = '' } =
+    currentArea || {};
+  const { name: mergeAreaName = undefined, ref: mergeAreaRef = '' } =
+    mergeArea || {};
```

Comment:
Please just give these guys a default {} rather than doing the null check like this